framework/src/modules/chain/modules/process_transactions.js

```diff
@@ -60,6 +60,37 @@ class ProcessTransactions {
 		};
 	}
 
+	// eslint-disable-next-line class-methods-use-this
+	async undoTransactions(transactions, tx = undefined) {
+		// Get data required for verifying transactions
+		const stateStore = library.logic.stateManager.createStore({
+			mutate: true,
+			tx,
+		});
+
+		const transactionInstances = transactions
+			.map(transaction => ({
+				...transaction,
+				fee: transaction.fee.toString(),
+				amount: transaction.amount.toString(),
+				recipientId: transaction.recipientId || '',
+			}))
+			.map(initTransaction);
+
+		await Promise.all(transactionInstances.map(t => t.prepare(stateStore)));
+
+		const transactionResponses = transactionInstances.map(transaction => {
+			const transactionResponse = transaction.undo(stateStore);
+			stateStore.transaction.add(transaction);
```

Comment:
I think in the undoConfirmedStep, all transactions should be added in the beginning and a particular transaction should be removed for which undo method is called.

Reply:
Changed the code so it does not add transactions to the store as the accounts are the ones that are mutated on this operation and transactions are deleted by the FK `trs_blockId_fkey` on the `trs` table.

Reply:
was previously deleted automatically by FK or explicitly calling delete?

Reply:
Yes it was automatically deleted by the FK already.